### PR TITLE
Use Out of band Cache instead of the one baked in

### PR DESCRIPTION
**Pull request by SivanMehta**
This stops re-inventing the caching wheel and uses `out-of-band-cache` instead. By changing the tests as little as possible I was able (I think) to ensure public API parity, but please let me know if the changes I made are breaking. However, because `out-of-band-cache` makes heavy use of `async/await`, this is entirely [broken in node@6](https://travis-ci.org/warehouseai/warehouse.ai-api-client/jobs/413690666#L1736), but not node@8

Edit: Will be a **major** bump because we break support for `node@6`

### CODE REVIEW

**Review by indexzero — CHANGES_REQUESTED** [review]
Please remove `node@6` from `.travis.yml` and add support for `node@10`

File: builds.js [hunk 1]
```diff
@@ -40,9 +40,8 @@ class Builds {
     this.warehouse = warehouse;
 
     options = Object.assign({}, defaultOptions, options);
-    if (options.cache && options.cache.enabled) {
-      this.cache = new Builds.Cache(this._get.bind(this), options.cache);
-    }
+    this.cache = new Cache({});
```
Comment (indexzero) [inline builds.js]: This accepts options. We should pass them through from the client (e.g. `fsCachePath`).

File: test/builds.test.js [hunk 2]
```diff
@@ -177,22 +174,6 @@ describe('Builds', function () {
...
-
-      builds.get({ pkg: 'some-pkg' }, (error, data) => {
-        assume(error).is.falsey();
-        assume(data).equals(buildData);
-        assume(sendStub).is.called(1);
-        assume(builds.cache._cache).is.truthy();
-        assume(builds.cache._cache.size).equals(1);
-        done();
-      });
-    });
```
Comment (indexzero) [inline test/builds.test.js]: This test feels still relevant.

**Review by indexzero — APPROVED** [review]
Given that this change drops support for `node@6` it is a breaking change, by design.

**Review by jcrugzz — APPROVED** [review]

**Pull request merged**